src/network/inet_aton.c

```diff
@@ -3,21 +3,31 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <stdlib.h>
+#pragma CHECKED_SCOPE on
 
-int __inet_aton(const char *s0, struct in_addr *dest)
+int __inet_aton(const char *s0 : itype(_Nt_array_ptr<const char>),
+	struct in_addr *dest : itype(_Ptr<struct in_addr>))
 {
-	const char *s = s0;
-	unsigned char *d = (void *)dest;
-	unsigned long a[4] = { 0 };
-	char *z;
+	_Nt_array_ptr<const char> s = s0;
+	_Array_ptr<unsigned char> d : count(4) = (_Array_ptr<unsigned char>)dest;
```

Comment:
Could you please add comments wherever you have plain integers as the bounds? Otherwise, it's difficult to reason about the bounds.

Reply:
Added. Thanks!